src/components/DatePicker/DatePickerTrigger.tsx

```diff
@@ -12,28 +13,74 @@ type DatePickerTriggerProps = PropsWithChildren<{
     isClearable?: boolean;
     isCalendarOpen?: boolean;
     onClick?: () => void;
+    validation?: Validation;
+    onDateChanged?: (date: Date | null) => void;
 }>;
 
 export const DatePickerTrigger = forwardRef<HTMLDivElement, DatePickerTriggerProps>(
-    ({ value, onClick, isClearable, placeHolder, isCalendarOpen }, ref) => (
-        <div onClick={onClick} ref={ref} aria-hidden="true">
-            <div
-                className={merge([
-                    'tw-absolute tw-top-2 tw-text-black-60 tw-z-10',
-                    value === '' || !isClearable ? 'tw-right-4' : 'tw-right-8',
-                ])}
-            >
-                {isCalendarOpen ? <IconCaretUp size={IconSize.Size20} /> : <IconCaretDown size={IconSize.Size20} />}
+    (
+        { value, onClick, isClearable, placeHolder, isCalendarOpen, validation = Validation.Default, onDateChanged },
+        ref,
+    ) => {
+        const isErrorValidationState = validation === Validation.Error;
+
+        const twoSlotsAwayFromRightEdge = 'tw-right-14';
+        const oneSlotAwayFromRightEdge = 'tw-right-8';
+        const zeroSlotsAwayFromRightEdge = 'tw-right-4';
+
+        const carrotRightSideTWPositionClassByOffset = (slotsOffsetFromRighSide: number): string => {
+            switch (slotsOffsetFromRighSide) {
+                case 0:
+                    return zeroSlotsAwayFromRightEdge;
+                case 1:
+                    return oneSlotAwayFromRightEdge;
+                case 2:
+                    return twoSlotsAwayFromRightEdge;
+            }
+            return '';
+        };
+
+        const carrotOffsetForClearButton = value && isClearable ? 1 : 0;
+        const carrotOffsetForValidationErrorIcon = isErrorValidationState ? 1 : 0;
+
+        const carrotRightSideTWPositionClass = carrotRightSideTWPositionClassByOffset(
+            carrotOffsetForClearButton + carrotOffsetForValidationErrorIcon,
+        );
+
+        /* eslint-disable jsx-a11y/click-events-have-key-events,jsx-a11y/no-static-element-interactions */
```

Comment:
There is no way we can fix this accessibility issue and not disable eslint? 🤔

Reply:
I wouldn't add any keyboard interaction to this element as it's already possible to open the the DatePicker by hitting enter when you focus the whole input. Making this element focusable would make it look strange.

The purpose of this element is to handle clicks for opening when the clear icon (or clear icon and error validation icon both ) are present, but leaving out space for the clear icon to be clickable as well.